### PR TITLE
Experimental work speed modifiers

### PR DESCRIPTION
**Pull request by tile4**
## About The Pull Request
Makes people who are below the stat average have a higher work speed and people way above it have a slower work speed.

## Why It's Good For The Game
It would make it easier for people behind to catch up to everybody else.

## Changelog
:cl:
add: Added new things
tweak: tweaked a few things
/:cl:

### CODE REVIEW

File: code/game/machinery/computer/abnormality_work.dm [hunk 1]
```diff
@@ -214,6 +214,43 @@
...
 	var/init_work_speed = work_speed
+	var/agent_stat_total = 0
+	var/work_agent_stat_total = 0
+	var/agent_count = 0
+	for(var/mob/living/carbon/human/H in AllLivingAgents())
+		if(!H.client)
+			continue
+		if(!H.mind)
+			continue
+		if(!H.z = z) //To prevent things like thunderdome from interfering from stat gain.
```
Comment (InsightfulParasite) [inline code/game/machinery/computer/abnormality_work.dm:225]: Ya gotta do == or else it causes intigration failures. I dont know why this isnt showing up as a linter issue.

Comment (tile4) [reply code/game/machinery/computer/abnormality_work.dm:225]: ...How did I manage to make that stupid mistake....

**Comment by vampirebat74** [comment]
Why is this necessary, exactly? This just punishes optimal play in general. There are going to be players who are either afk, or completely new to the game no matter what, and this just gives another reason for them to be harassed and bullied.

**Comment by tile4** [comment]
> Why is this necessary, exactly? This just punishes optimal play in general. There are going to be players who are either afk, or completely new to the game no matter what, and this just gives another reason for them to be harassed and bullied.

The numbers are temporary, but I am just trying to make ways for people to be able to keep up when nobody gives them slack, and also kinda punishes people for powergrinding to absurd levels. if I need to adjust the exact values, I can. But I just wanted to at least establish the concept of my idea first.

**Comment by Litberries** [comment]
> > Why is this necessary, exactly? This just punishes optimal play in general. There are going to be players who are either afk, or completely new to the game no matter what, and this just gives another reason for them to be harassed and bullied.
> 
> The numbers are temporary, but I am just trying to make ways for people to be able to keep up when nobody gives them slack, and also kinda punishes people for powergrinding to absurd levels. if I need to adjust the exact values, I can. But I just wanted to at least establish the concept of my idea first.

this also punishes players who just want to play the game at a normal pace due to those lagging behind I'm pretty sure

**Comment by vampirebat74** [comment]
> > Why is this necessary, exactly? This just punishes optimal play in general. There are going to be players who are either afk, or completely new to the game no matter what, and this just gives another reason for them to be harassed and bullied.
> 
> The numbers are temporary, but I am just trying to make ways for people to be able to keep up when nobody gives them slack, and also kinda punishes people for powergrinding to absurd levels. if I need to adjust the exact values, I can. But I just wanted to at least establish the concept of my idea first.

Punishing agents for doing their one and only role is rather short-sighted.

**Comment by tile4** [comment]
> > > Why is this necessary, exactly? This just punishes optimal play in general. There are going to be players who are either afk, or completely new to the game no matter what, and this just gives another reason for them to be harassed and bullied.
> > 
> > 
> > The numbers are temporary, but I am just trying to make ways for people to be able to keep up when nobody gives them slack, and also kinda punishes people for powergrinding to absurd levels. if I need to adjust the exact values, I can. But I just wanted to at least establish the concept of my idea first.
> 
> Punishing agents for doing their one and only role is rather short-sighted.

The goal is to punish the ones that overperform to a extreme amount. I can change the numbers if they are too harsh.

**Comment by tile4** [comment]
> > > Why is this necessary, exactly? This just punishes optimal play in general. There are going to be players who are either afk, or completely new to the game no matter what, and this just gives another reason for them to be harassed and bullied.
> > 
> > 
> > The numbers are temporary, but I am just trying to make ways for people to be able to keep up when nobody gives them slack, and also kinda punishes people for powergrinding to absurd levels. if I need to adjust the exact values, I can. But I just wanted to at least establish the concept of my idea first.
> 
> this also punishes players who just want to play the game at a normal pace due to those lagging behind I'm pretty sure
With how I am trying to impliment it, it really shouldn't unless you got a lot of people lagging behind...

**Comment by Coxswain-Navigator** [comment]
Why punish players for succeeding? It would make more sense to help the struggling agents directly.

**Comment by tile4** [comment]
> Why punish players for succeeding? It would make more sense to help the struggling agents directly.

My goal isn't to punish players for succeeding, but over succeeding that usually occurs from more powergrindy behaviors I guess. Maybe this isn't the best way to get this accomplished but it is hard to try to force teamwork (which is like the only way to get some of the agents to work together, without harming low pop too much. So I got this idea I guess.

**Comment by tile4** [comment]
You know what? I think I am stupid.... I forgot the bigger issue to tackle rather than punishing those people oversucceeding...

**Comment by Halonexus** [comment]
I think the helping people who are lagging behind is part of this is fine

**Pull request closed without merging**